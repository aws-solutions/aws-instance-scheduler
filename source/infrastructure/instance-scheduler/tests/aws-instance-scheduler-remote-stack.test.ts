// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { createSpokeStack } from "./instance-scheduler-stack-factory";
import { Template } from "aws-cdk-lib/assertions";

/*
 * SnapShot Testing for the AwsInstanceSchedulerStack.
 */
test("AwsInstanceSchedulerStack snapshot test", () => {
  expect(Template.fromStack(createSpokeStack())).toMatchSnapshot();
});
