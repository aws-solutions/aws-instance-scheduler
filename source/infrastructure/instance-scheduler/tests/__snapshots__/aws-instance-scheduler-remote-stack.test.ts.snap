// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwsInstanceSchedulerStack snapshot test 1`] = `
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": "",
  "Mappings": {
    "AppRegistryForInstanceSchedulerSolution25A90F05": {
      "Data": {
        "AppRegistryApplicationName": "instance-scheduler-on-aws",
        "ApplicationType": "AWS-Solutions",
        "ID": "SO0030",
        "SolutionName": "aws-instance-scheduler",
        "Version": "v1.5.0",
      },
    },
  },
  "Metadata": {
    "AWS::CloudFormation::Interface": {
      "ParameterGroups": [
        {
          "Label": {
            "default": "Account",
          },
          "Parameters": [
            "InstanceSchedulerAccount",
          ],
        },
      ],
      "ParameterLabels": {
        "InstanceSchedulerAccount": {
          "default": "Primary account",
        },
      },
    },
  },
  "Outputs": {
    "AppRegistryApplicationManagerUrl775D5C3D": {
      "Description": "Application manager url for the application created.",
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://eu-west-1.console.aws.amazon.com/systems-manager/appmanager/application/AWS_AppRegistry_Application-",
            {
              "Fn::Join": [
                "-",
                [
                  {
                    "Fn::FindInMap": [
                      "AppRegistryForInstanceSchedulerSolution25A90F05",
                      "Data",
                      "AppRegistryApplicationName",
                    ],
                  },
                  {
                    "Ref": "AWS::Region",
                  },
                  {
                    "Ref": "AWS::AccountId",
                  },
                  {
                    "Ref": "AWS::StackName",
                  },
                ],
              ],
            },
          ],
        ],
      },
    },
    "CrossAccountRole": {
      "Description": "Arn for cross account role for Instance scheduler, add this arn to the list of crossaccount roles (CrossAccountRoles) parameter of the Instance Scheduler template.",
      "Value": {
        "Fn::GetAtt": [
          "EC2SchedulerCrossAccountRole",
          "Arn",
        ],
      },
    },
  },
  "Parameters": {
    "InstanceSchedulerAccount": {
      "AllowedPattern": "(^[0-9]{12}$)",
      "ConstraintDescription": "Account number is a 12 digit number",
      "Description": "Account number of Instance Scheduler account to give access to manage EC2 and RDS  Instances in this account.",
      "Type": "String",
    },
  },
  "Resources": {
    "AppRegistry968496A3": {
      "Properties": {
        "Description": {
          "Fn::Join": [
            "",
            [
              "Service Catalog application to track and manage all your resources for the solution ",
              {
                "Fn::FindInMap": [
                  "AppRegistryForInstanceSchedulerSolution25A90F05",
                  "Data",
                  "SolutionName",
                ],
              },
            ],
          ],
        },
        "Name": {
          "Fn::Join": [
            "-",
            [
              {
                "Fn::FindInMap": [
                  "AppRegistryForInstanceSchedulerSolution25A90F05",
                  "Data",
                  "AppRegistryApplicationName",
                ],
              },
              {
                "Ref": "AWS::Region",
              },
              {
                "Ref": "AWS::AccountId",
              },
              {
                "Ref": "AWS::StackName",
              },
            ],
          ],
        },
        "Tags": {
          "Solutions:ApplicationType": {
            "Fn::FindInMap": [
              "AppRegistryForInstanceSchedulerSolution25A90F05",
              "Data",
              "ApplicationType",
            ],
          },
          "Solutions:SolutionID": {
            "Fn::FindInMap": [
              "AppRegistryForInstanceSchedulerSolution25A90F05",
              "Data",
              "ID",
            ],
          },
          "Solutions:SolutionName": {
            "Fn::FindInMap": [
              "AppRegistryForInstanceSchedulerSolution25A90F05",
              "Data",
              "SolutionName",
            ],
          },
          "Solutions:SolutionVersion": {
            "Fn::FindInMap": [
              "AppRegistryForInstanceSchedulerSolution25A90F05",
              "Data",
              "Version",
            ],
          },
        },
      },
      "Type": "AWS::ServiceCatalogAppRegistry::Application",
    },
    "AppRegistryAttributeGroupAssociation2328adb27afd49C57352": {
      "Properties": {
        "Application": {
          "Fn::GetAtt": [
            "AppRegistry968496A3",
            "Id",
          ],
        },
        "AttributeGroup": {
          "Fn::GetAtt": [
            "AppRegistryForInstanceSchedulerDefaultApplicationAttributes78274898",
            "Id",
          ],
        },
      },
      "Type": "AWS::ServiceCatalogAppRegistry::AttributeGroupAssociation",
    },
    "AppRegistryForInstanceSchedulerDefaultApplicationAttributes78274898": {
      "Properties": {
        "Attributes": {
          "applicationType": {
            "Fn::FindInMap": [
              "AppRegistryForInstanceSchedulerSolution25A90F05",
              "Data",
              "ApplicationType",
            ],
          },
          "solutionID": {
            "Fn::FindInMap": [
              "AppRegistryForInstanceSchedulerSolution25A90F05",
              "Data",
              "ID",
            ],
          },
          "solutionName": {
            "Fn::FindInMap": [
              "AppRegistryForInstanceSchedulerSolution25A90F05",
              "Data",
              "SolutionName",
            ],
          },
          "version": {
            "Fn::FindInMap": [
              "AppRegistryForInstanceSchedulerSolution25A90F05",
              "Data",
              "Version",
            ],
          },
        },
        "Description": "Attribute group for solution information",
        "Name": {
          "Fn::Join": [
            "-",
            [
              {
                "Ref": "AWS::Region",
              },
              {
                "Ref": "AWS::StackName",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ServiceCatalogAppRegistry::AttributeGroup",
    },
    "AppRegistryResourceAssociationdf1bce20c35e43B5355E": {
      "Properties": {
        "Application": {
          "Fn::GetAtt": [
            "AppRegistry968496A3",
            "Id",
          ],
        },
        "Resource": {
          "Ref": "AWS::StackId",
        },
        "ResourceType": "CFN_STACK",
      },
      "Type": "AWS::ServiceCatalogAppRegistry::ResourceAssociation",
    },
    "EC2SchedulerCrossAccountRole": {
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-IAM5",
              "reason": "All policies have been scoped to be as restrictive as possible. This solution needs to access ec2/rds resources across all regions.",
            },
          ],
        },
        "cfn_nag": {
          "rules_to_suppress": [
            {
              "id": "W11",
              "reason": "All policies have been scoped to be as restrictive as possible. This solution needs to access ec2/rds resources across all regions.",
            },
          ],
        },
      },
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Sub": [
                    "arn:\${AWS::Partition}:iam::\${accountId}:root",
                    {
                      "accountId": {
                        "Ref": "InstanceSchedulerAccount",
                      },
                    },
                  ],
                },
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Path": "/",
        "Policies": [
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": [
                    "rds:DeleteDBSnapshot",
                    "rds:DescribeDBSnapshots",
                    "rds:StopDBInstance",
                  ],
                  "Effect": "Allow",
                  "Resource": {
                    "Fn::Sub": "arn:\${AWS::Partition}:rds:*:\${AWS::AccountId}:snapshot:*",
                  },
                },
                {
                  "Action": [
                    "rds:AddTagsToResource",
                    "rds:RemoveTagsFromResource",
                    "rds:DescribeDBSnapshots",
                    "rds:StartDBInstance",
                    "rds:StopDBInstance",
                  ],
                  "Effect": "Allow",
                  "Resource": {
                    "Fn::Sub": "arn:\${AWS::Partition}:rds:*:\${AWS::AccountId}:db:*",
                  },
                },
                {
                  "Action": [
                    "rds:AddTagsToResource",
                    "rds:RemoveTagsFromResource",
                    "rds:StartDBCluster",
                    "rds:StopDBCluster",
                  ],
                  "Effect": "Allow",
                  "Resource": {
                    "Fn::Sub": "arn:\${AWS::Partition}:rds:*:\${AWS::AccountId}:cluster:*",
                  },
                },
                {
                  "Action": [
                    "ec2:StartInstances",
                    "ec2:StopInstances",
                    "ec2:CreateTags",
                    "ec2:DeleteTags",
                  ],
                  "Effect": "Allow",
                  "Resource": {
                    "Fn::Sub": "arn:\${AWS::Partition}:ec2:*:\${AWS::AccountId}:instance/*",
                  },
                },
                {
                  "Action": [
                    "rds:DescribeDBClusters",
                    "rds:DescribeDBInstances",
                    "ec2:DescribeInstances",
                    "ec2:DescribeRegions",
                    "ssm:DescribeMaintenanceWindows",
                    "ssm:DescribeMaintenanceWindowExecutions",
                    "tag:GetResources",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "EC2InstanceSchedulerRemote",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "Ec2ModifyInstanceAttrPolicy4B693ACF": {
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-IAM5",
              "reason": "All policies have been scoped to be as restrictive as possible. This solution needs to access ec2/rds resources across all regions.",
            },
          ],
        },
      },
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "ec2:ModifyInstanceAttribute",
              "Effect": "Allow",
              "Resource": {
                "Fn::Sub": "arn:\${AWS::Partition}:ec2:*:\${AWS::AccountId}:instance/*",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "Ec2ModifyInstanceAttrPolicy4B693ACF",
        "Roles": [
          {
            "Ref": "EC2SchedulerCrossAccountRole",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
